rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactory.java

```diff
@@ -39,4 +48,24 @@ protected String getBarcodeValsAsCSV(Intent data) {
     protected String[] splitCSV(String barcodeCSV) {
         return barcodeCSV.split(",");
     }
-}
+
+    @Override
+    protected void moveToNextStep(Intent data, Date expDate) {
+        final int sensorTriggerIndex = 4;
+        Barcode barcode = data.getParcelableExtra(BARCODE_KEY);
+        if (Boolean.parseBoolean(barcode.displayValue.split(",")[sensorTriggerIndex])) {
+            JsonApi jsonApi = (JsonApi) widgetArgs.getContext();
+            String expiredPageAddr = stepStateConfig.getStepStateObj().optString(Constants.Step.PRODUCT_EXPIRED_PAGE, "step1");
+            try {
+                jsonApi.writeValue(expiredPageAddr, "lbl_sample_expired", "",  "", "", "", false);
+                jsonApi.writeValue(expiredPageAddr, "lbl_collect_new_sample", widgetArgs.getContext().getString(R.string.rdt_high_heat_exposure),  "", "", "", false);
+            } catch (JSONException ex) {
+                Timber.e(ex);
+            }
+            JsonFormFragment nextFragment = RDTJsonFormFragment.getFormFragment(expiredPageAddr);
+            widgetArgs.getFormFragment().transactThis(nextFragment);
```

Comment:
This is the same logic applied here https://github.com/onaio/rdt-standard/blob/9c36969c9f51552254872699fe1613d6062af1ac/rdt-open-reader/src/main/java/io/ona/rdt/widget/RDTBarcodeFactory.java#L97.

Why not refactor that logic into a method with signature `moveToNextStep(boolean isUsableProduct)` which moves to next step if product is usable and navigates to `PRODUCT_EXPIRED_PAGE` otherwise.

Then we can call the method here with the value of `sensorTriggered`.

Reply:
you mean to create a new method name `moveToNextStep(boolean isUsableProduct)` inside an `RDTBarcodeFactor.java` class and if so then where can I get the `isUsableProduct` variable from.